core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala

```diff
@@ -304,22 +304,4 @@ class SparkSubmitUtilsSuite extends SparkFunSuite with BeforeAndAfterAll {
         s" Resolved jars are: $jarPath")
     }
   }
-
-  test("SPARK-34757: should ignore cache for SNAPSHOT dependencies") {
```

Comment:
@bozhang2820, is the issue only in the test? I think the change in the main code is correct. If you manually tested and confirmed that it works, let's just remove the test only for now (as a followup of SPARK-34757). I think that's fine.